In [1]:
from scipy.stats import norm

# Calculating the size that the sample should be

Let's assume that we want a maximum error of 0.0001, a 95% confidence level and the population standard deviation is 0.0019.

\begin{equation}
e = z \frac{\sigma}{\sqrt{n}}
\end{equation}


known standard deviation:
\begin{equation}
n = \left( z \frac{\sigma}{e} \right)^2
\end{equation}


unknown standard deviation:
\begin{equation}
n = \left( z \frac{s}{e} \right)^2
\end{equation}

In [5]:
#area under the curve

0.95 + 0.025

0.975

In [6]:
# getting z value
z = norm.ppf(0.975)

z

1.959963984540054

In [7]:
sigma = 0.0019
error = 0.0001

In [8]:
n = (z * sigma / error)**2

n

1386.766634270579

- The sample size should be 1386.766634270579

What If my population is less than 1386?

## Finite Population

\begin{equation}
n = \frac{z^2 \sigma^2 N}{z^2 \sigma^2 + e^2(N-1)}
\end{equation}

In [10]:
# N is the maximum number of the population

N = 1500

In [11]:
((z**2) * (sigma**2) * N) / ((z**2) * (sigma**2) + (error**2) * (N-1))

720.8309662682263

- In this case, that my the size of the population is 1500, the sample size should be 720.8309662682263.